### DISCUSSION TITLE
Typst – keep templates as external files rather than injecting it in the ouput

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#9514] [Feature Requests]
### Description

When using a custom template, the current strategy copies over the template into the output document, making it harder to post-process when part of a toolchain (e.g. one may want to `keep-typ: true` and manually edit the output – having all the template content in there makes for a difficult editing experience). This would be similar to Latex in that the style file is defined in a separate file.

**Comment by mcanouil** [comment]
Do you mean partials? (https://github.com/quarto-dev/quarto-cli/issues/8500)

**Reply by cscheid** [reply]
No, I think they mean that we should actually use the notion of a _typst_ template (https://typst.app/docs/tutorial/making-a-template/) rather than a pandoc template, which would clean up the `.typ` file we generate.

> This would be similar to Latex in that the style file is defined in a separate file.

Do note that our .tex file is not nearly as clean as you might want, and for the same reason.

In other words, this is not currently a priority for us. We take the .tex and .typ files to be debugging and production aids rather than files we expect people to edit. You're of course welcome to edit it if necessary, but you're already signing up for navigating code-generated output. It's not currently worth our engineering efforts to clean up some of this stuff. We prefer to make it so that people don't need to edit the files to begin with.

**Reply by baptiste** [reply]
Yes, that's what I meant. Fair enough, I understand it's not a priority. 

For documents that don't have embedded computations (assignments, exams, flyers, general admin stuff, etc.), I'm finding more and more that I [prefer to author in Typst](https://github.com/quarto-dev/quarto-cli/discussions/5024#discussioncomment-5470233) rather than markdown, as it's [more expressive yet still succinct](https://docs.racket-lang.org/pollen/second-tutorial.html#%28part._the-case-against-markdown%29), and I can just pass those files via the online app to colleagues who're never going to install computational software to typeset something. I can't give them the current typst output because it's prefaced by many lines of incomprehensible template stuff.

In one project I have ~30 qmd files with a custom typst template that I was hoping to convert automatically to typst (body and yaml). But for this one-off I can probably just hack something together with glue().
